kubernetes/resource_kubernetes_pod_security_policy.go

```diff
@@ -0,0 +1,494 @@
+	pspSpecPrivilegedDoc                      = api.PodSecurityPolicySpec{}.SwaggerDoc()["privileged"]
+	pspSpecReadOnlyRootFilesystemDoc          = api.PodSecurityPolicySpec{}.SwaggerDoc()["readOnlyRootFilesystem"]
+	pspSpecRequiredDropCapabilitiesDoc        = api.PodSecurityPolicySpec{}.SwaggerDoc()["requiredDropCapabilities"]
+	pspSpecRunAsUserDoc                       = api.PodSecurityPolicySpec{}.SwaggerDoc()["runAsUser"]
+	pspRunAsUserIDRangeDoc                    = api.RunAsUserStrategyOptions{}.SwaggerDoc()["ranges"]
+	pspRunAsUserRuleDoc                       = api.RunAsUserStrategyOptions{}.SwaggerDoc()["rule"]
+	pspSpecSELinuxDoc                         = api.PodSecurityPolicySpec{}.SwaggerDoc()["seLinux"]
+	pspSELinuxOptionsDoc                      = api.SELinuxStrategyOptions{}.SwaggerDoc()["seLinuxOptions"]
+	pspSELinuxOptionsLevelDoc                 = api.SELinuxStrategyOptions{}.SwaggerDoc()["level"]
+	pspSELinuxOptionsRoleDoc                  = api.SELinuxStrategyOptions{}.SwaggerDoc()["role"]
+	pspSELinuxOptionsTypeDoc                  = api.SELinuxStrategyOptions{}.SwaggerDoc()["type"]
+	pspSELinuxOptionsUserDoc                  = api.SELinuxStrategyOptions{}.SwaggerDoc()["user"]
+	pspSELinuxOptionsRuleDoc                  = api.SELinuxStrategyOptions{}.SwaggerDoc()["rule"]
+	pspSpecSupplementalGroupsDoc              = api.PodSecurityPolicySpec{}.SwaggerDoc()["supplementalGroups"]
+	pspSupplementalGroupsRangesDoc            = api.SupplementalGroupsStrategyOptions{}.SwaggerDoc()["ranges"]
+	pspSupplementalGroupsRuleDoc              = api.SupplementalGroupsStrategyOptions{}.SwaggerDoc()["rule"]
+	pspSpecVolumesDoc                         = api.PodSecurityPolicySpec{}.SwaggerDoc()["volumes"]
+	pspSpecRunAsGroupDoc                      = api.PodSecurityPolicySpec{}.SwaggerDoc()["runAsGroup"]
+	pspRunAsGroupIDRangeDoc                   = api.RunAsGroupStrategyOptions{}.SwaggerDoc()["ranges"]
+	pspRunAsGroupRuleDoc                      = api.RunAsGroupStrategyOptions{}.SwaggerDoc()["rule"]
+)
+
+func resourceKubernetesPodSecurityPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKubernetesPodSecurityPolicyCreate,
+		Read:   resourceKubernetesPodSecurityPolicyRead,
+		Exists: resourceKubernetesPodSecurityPolicyExists,
+		Update: resourceKubernetesPodSecurityPolicyUpdate,
+		Delete: resourceKubernetesPodSecurityPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"metadata": metadataSchema("podsecuritypolicy", false),
+			"spec": {
+				Type:        schema.TypeList,
+				Description: pspSpecDoc,
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"allow_privilege_escalation": {
+							Type:        schema.TypeBool,
+							Description: pspSpecAllowPrivilegeEscalationDoc,
+							Optional:    true,
+							Computed:    true,
```

Comment:
Hi, out of curiosity, why Computed? I assume it may be for an import operation or to assume an API provided default?

Thank you,
Ed

Reply:
This is me being lazy and not wanting to codify the api default behaviour in terraform. If this value is not set then the policy api will default it to true (and the returned psp will have the value set). I figured it was safer to let the API handle setting defaults and just tell the terraform schema that these values could be computed or specified.

Reply:
Thanks for the info!